intermediate_source/realtime_rpi.rst

```diff
@@ -0,0 +1,259 @@
+
+We then want to jit the model to reduce Python overhead and fuse any ops. Jit gives us ~30fps instead of ~20fps without it.
+
+.. code:: python
+
+    net = torch.jit.script(net)
+    net.eval()
+
+Putting It Together
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We can now put all the pieces together and run it:
+
+.. code:: python
+
+    import time
+
+    import torch
+    import numpy as np
+    from torchvision import models, transforms
+
+    import cv2
+    from PIL import Image
+
+    torch.backends.quantized.engine = 'qnnpack'
+
+    cap = cv2.VideoCapture(0, cv2.CAP_V4L2)
+    cap.set(cv2.CAP_PROP_FRAME_WIDTH, 224)
+    cap.set(cv2.CAP_PROP_FRAME_HEIGHT, 224)
+    cap.set(cv2.CAP_PROP_FPS, 36)
+
+    preprocess = transforms.Compose([
+        transforms.ToTensor(),
+        transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
+    ])
+
+    net = models.quantization.mobilenet_v2(pretrained=True, quantize=True)
+    # jit model to take it from ~20fps to ~30fps
+    net = torch.jit.script(net)
+    net.eval()
+
+    started = time.time()
+    last_logged = time.time()
+    frame_count = 0
+
+    with torch.no_grad():
+        while True:
+            # read frame
+            ret, image = cap.read()
+            if not ret:
+                raise RuntimeError("failed to read frame")
+
+            # convert opencv output from BGR to RGB
+            image = image[:, :, [2, 1, 0]]
+            permuted = image
+
+            # preprocess
+            input_tensor = preprocess(image)
+
+            # create a mini-batch as expected by the model
+            input_batch = input_tensor.unsqueeze(0)
+
+            # run model
+            output = net(input_batch)
+            # do something with output ...
+
+            # log model performance
+            frame_count += 1
+            now = time.time()
+            if now - last_logged > 1:
+                print(f"{frame_count / (now-last_logged)} fps")
+                last_logged = now
+                frame_count = 0
+
+Running it shows that we're hovering at ~30 fps.
+
+.. image:: https://user-images.githubusercontent.com/909104/152892609-7d115705-3ec9-4f8d-beed-a51711503a32.png
+
+This is with all the default settings in Raspberry Pi OS. If you disabled the UI
+and all the other background services that are enabled by default it's more
+performant and stable.
+
+If we check ``htop`` we see that we have almost 100% utilization.
+
+.. image:: https://user-images.githubusercontent.com/909104/152892630-f094b84b-19ba-48f6-8632-1b954abc59c7.png
+
+Next Steps
+~~~~~~~~~~~~~
+
+You can create your own model or fine tune an existing one. If you fine tune on
+one of the models from
+`torchvision.models.quantized
+<https://pytorch.org/vision/stable/models.html#quantized-models>`_
+most of the work to fuse and quantize has already been done for you so you can
+directly deploy with good performance on a Raspberry Pi.
+
+See more:
```

Comment:
I'd like to see a gif of the model working, you can even showcase what it's seeing - will also be nice to share in tweet threads

Reply:
yup, definitely planning on it. no more mysterious gdrive python packages